File: lab_5/lab_5(java)/src/Series/GeoProg.java
package Series;

public class GeoProg extends Series{
    public GeoProg(double first, double step, int size) {
        super(first, step, size);
    }

    public GeoProg() {
        super(1, 2, 5);
    }

    @Override
    public double getElement(int index) {
        double element = this.first * Math.pow(this.step, index-1);
        return element;
    }

    @Override
    public double getSum() {
        return this.first * ( 1 - Math.pow(this.step, this.size) / (1 - this.step));
    }
}

File: lab_5/Lab_5/Series/GeoProg.cpp
#include "GeoProg.hpp"

GeoProg::GeoProg(double first, double step, int size) : Series(first, step, size) {}

GeoProg::~GeoProg() {
        std::cout << "GeoProg::~GeoProg()" << std::endl;
}

double GeoProg::getElement(int index) {
    double element = this->first * pow(this->step, index-1);
    return element;
}

double GeoProg::getSum() {
    return this->first * ( 1 - pow(this->step, this->size) / (1 - this->step));
}
